mantidproject/mantid
# AppendSpectra now properly sets the values of TextAxis and NumericAxis

Description of work.

`AppendSpectra` algorithm used to ignore the values of the TextAxis and NumericAxis labels and just append the spectra number. This is fixed in this PR and the algorithm now properly appends the value of TextAxis and NumericAxis, while keeping the old behaviour unchanged.

**To test:**
- I've made a quite long testing script, that I used to test all cases I could think of, it can be found here: https://gist.github.com/DimitarTasev/1c16fbfe60b95f8dff4845fbb7b02393
- Run unit tests and make sure they pass successfully
- Code review, changes aren't that big, but if you can think of any better way to handle the unique cases, please tell me



Fixes #17902 .



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
